cmd/applyconfig/applyconfig.go

```diff
@@ -239,6 +242,55 @@ func applyConfig(rootDir, cfgType string, process processFn) error {
 	return nil
 }
 
+type censoringFormatter struct {
+	secrets  sets.String
+	delegate logrus.Formatter
+}
+
+func (f censoringFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	raw, err := f.delegate.Format(entry)
+	if err != nil {
+		return raw, err
+	}
+
+	return f.Censor(raw), nil
+}
+
+const censored = "CENSORED"
+
+var censoredBytes = []byte(censored)
+
+func (f *censoringFormatter) Censor(content []byte) []byte {
+	if f.secrets != nil {
+		for _, secret := range f.secrets.List() {
+			content = bytes.ReplaceAll(content, []byte(secret), censoredBytes)
+		}
+	}
+	return content
+}
+
+func getCensoringFormatter(f logrus.Formatter) logrus.Formatter {
+	secrets := sets.NewString()
+	//those strings do not look like valid secrets
+	noSecrets := sets.NewString("", ":", "1")
+	for _, e := range os.Environ() {
```

Comment:
Huh this looks excessive to me... Have you considered adding the value to the list of items to be censored only after we actually know we read it and will use it as a template parameter?

Reply:
That was my original try but gave up after talking to you via [slack](https://coreos.slack.com/archives/GB7NB0CUC/p1571403302269400?thread_ts=1571403013.266700&cid=GB7NB0CUC).

Will try to use the params only.

Reply:
Sorry for the confusion... I meant _"whatever we read"_ in that post like "whatever we actually read and use from env var" :(

Reply:
Done.

@petr-muller 
I am thinking about the comments from @stevekuznetsov above:

I think the censoring logger here is more generic than the one used in `secretAgent` of `test-infra`.
Do you think it is worth a try at upstream (with necessary public class and function) so that `secretAgent` and `applyConfig` can use the shared code via `import`?

Reply:
Yeah, definitely. There's even a good place for it already, I think: https://github.com/kubernetes/test-infra/tree/master/prow/logrusutil